src/probnum/diffeq/wrappedscipysolver.py

```diff
@@ -0,0 +1,127 @@
+        self.solver.y = self.ivp.initrv.mean
+        self.solver.f = self.solver.fun(self.solver.t, self.solver.y)
+        return self.ivp.t0, self.ivp.initrv
+
+    def step(self, start: float, stop: float, current: randvars, **kwargs):
+        """Perform one ODE-step from start to stop and set variables to the
+        corresponding values.
+
+        To specify start and stop directly, rk_step() and not _step_impl() is used.
+
+        Parameters
+        ----------
+        start : float
+            starting location of the step
+        stop : float
+            stopping location of the step
+        current : :obj:`list` of :obj:`RandomVariable`
+            current state of the ODE.
+
+        Returns
+        -------
+        random_var : randvars.RandomVariable
+            Estimated states (in the state-space model view) of the discrete-time solution.
+        error_estimation : float
+            estimated error after having performed the step.
+        """
+
+        y = current.mean
+        dt = stop - start
+        y_new, f_new = rk.rk_step(
+            self.solver.fun,
+            start,
+            y,
+            self.solver.f,
+            dt,
+            self.solver.A,
+            self.solver.B,
+            self.solver.C,
+            self.solver.K,
+        )
+
+        # Unnormalized error estimation is used as the error estimation is normalized in
+        # solve().
+        error_estimation = self.solver._estimate_error(self.solver.K, dt)
+        y_new_as_rv = randvars.Constant(y_new)
+
+        # Update the solver settings. This part is copied from scipy's _step_impl().
+        self.solver.h_previous = dt
+        self.solver.y_old = current
+        self.solver.t_old = start
+        self.solver.t = stop
+        self.solver.y = y_new
+        self.solver.h_abs = dt
+        self.solver.f = f_new
+        return y_new_as_rv, error_estimation
+
+    def rvlist_to_odesol(self, times: np.array, rvs: np.array):
+        """Create a ScipyODESolution object which is a subclass of
+        diffeq.ODESolution."""
+        scipy_solution = OdeSolution(times, self.interpolants)
+        probnum_solution = wrappedscipyodesolution.WrappedScipyODESolution(
+            scipy_solution, rvs
+        )
+        return probnum_solution
+
+    def method_callback(self, time, current_guess, current_error):
+        """Call dense output after each step and store the interpolants."""
+        dense = self.dense_output()
+        self.interpolants.append(dense)
+
+    def dense_output(self):
+        """Compute the interpolant after each step with a quartic interpolation
+        polynomial.
+
+        Returns
+        -------
+        sol : rk.RkDenseOutput
+            interpolant between the last and current location.
+        """
+
+        Q = self.solver.K.T.dot(self.solver.P)
+        sol = rk.RkDenseOutput(
+            self.solver.t_old, self.solver.t, self.solver.y_old.mean, Q
+        )
+        return sol
```

Comment:
Yes, I first implemented this by testing whether the solver is an instance of RK23 or RK45, but that would mean that other (self-implemented) RK-methods can't use the dense output. I decided to write a comment about the methods that can be used without limitations and the not implemented DOP853 dense output in the class docstring. Please let me know if you have a better idea.

Reply:
Okay, thanks for clarifying :) However, was actually *happens* when one uses DOP853 in here? Does this still work, but is suboptimal in terms of speed (because it ignores the first-same-as-last property of DOP853), or does it break? What would a user expect? A simple solution to avoid these issues would be to check for DOP853 (as in: `if isinstance(self.solver, scipy.DOP853): raise TypeError` (which should probably happen at the init, not here?)) in which case the usage of these functions would be restricted to RK45, RK23, and whichever RK methods one decides to implement :)

Reply:
I added this check and included a test for it.

Reply:
wonderful!